src/thorin/be/llvm/llvm.cpp

```diff
@@ -888,6 +900,74 @@ llvm::Value* CodeGen::emit_lea(const LEA* lea) {
     return irbuilder_.CreateInBoundsGEP(lookup(lea->ptr()), args);
 }
 
+llvm::Value* CodeGen::emit_asm(const Assembly* assembly) {
+    const TupleType *out_type = assembly->type();
+    llvm::Type *res_type;
+    switch (out_type->num_args()) {
+        case 0:
+            THORIN_UNREACHABLE;
+            // there must always be the mem type as output
+        case 1:
+            res_type = llvm::Type::getVoidTy(context_);
+            break;
+        case 2:
+            res_type = convert(assembly->out(1)->type());
+            break;
+        default:
+            //res_type = convert(out_type);
+            // TODO: the llvm type that comes out of this is not added to the type_
+            // map, can this be a problem?
+            res_type = convert_tuple(out_type->args().begin(),
+                    out_type->args().end(), out_type->num_args());
+            break;
+    }
+
+    // TODO: better way to get the input params and types?
+    auto ops = assembly->ops();
+    auto op_size = ops.size() - 1;       // omit the mem type
+    auto input_params = new llvm::Value *[op_size];
+    auto param_types = new llvm::Type *[op_size];
```

Comment:
I can't use `thorin::Array` because I need to pass `input_params` and `param_types` to LLVM functions which don't accept it. I will use `std::vector` instead.